File: sample.cc
{
  gSystem->Load("libRooFit.so");
  gSystem->Load("libCommonFitting.so");

  gStyle->SetOptStat(0);

  TF1 landau("landau", "TMath::Landau(x, -10., 1.)", -20., 20.);
  TF1 gauss("gauss", "gaus(0)", -20., 20.);
  gauss.SetParameters(1., 0., 2.);

  TH1F* target = new TH1F("target", "Fit Target", 40, -20., 20.);
  TH1F* temp1 = new TH1F("temp1", "Template 1", 40, -20., 20.);
  TH1F* temp2 = new TH1F("temp2", "Template 2", 40, -20., 20.);

  target->FillRandom("landau", 2000);
  target->FillRandom("gauss", 1000);

  temp1->FillRandom("landau", 5000);

  temp2->FillRandom("gauss", 10000);

  TemplateChi2Fitter* fitter = TemplateChi2Fitter::singleton();

  fitter->setTarget(target);
  fitter->addTemplate(temp1, "temp1");
  fitter->addTemplate(temp2, "temp2");

  fitter->fit();

  double s1 = fitter->getFloat("temp1")->getVal();
  double s2 = fitter->getFloat("temp2")->getVal();

  temp1->Scale(s1);
  temp2->Scale(s2);

  temp1->SetLineColor(kRed);
  temp1->SetLineStyle(kDashed);

  temp2->SetLineColor(kBlue);
  temp2->SetLineStyle(kDashed);

  target->SetLineColor(kBlack);

  TCanvas* c1 = new TCanvas("c1", "c1");

  target->Draw();
  temp1->Draw("SAME");
  temp2->Draw("SAME");

  c1->Print("test.pdf");
}

File: sample.py
import ROOT
import array

ROOT.gSystem.Load('libRooFit.so')
ROOT.gSystem.Load('libFitTools.so')

fitter = ROOT.EfficiencyFitter.singleton()

tree = ROOT.TTree('test', 'test')
x = array.array('d', [0.])
p = array.array('I', [0])
tree.Branch('x', x, 'x/D')
tree.Branch('p', p, 'p/i')

data = []
data += [(0., 1)] * 4
data += [(0., 0)] * 1
data += [(0.5, 1)] * 3
data += [(0.5, 0)] * 2
data += [(1., 1)] * 2
data += [(1., 0)] * 3

for datum in data:
    x[0] = datum[0]
    p[0] = datum[1]
    tree.Fill()

xvar = ROOT.RooRealVar('x', 'x', -2., 2.)
intercept = ROOT.RooRealVar('intercept', 'intercept', 0.2, -10., 10.)
slope = ROOT.RooRealVar('slope', 'slope', 0.5, -1., 1.)
line = ROOT.RooPolyVar('line', 'line', xvar, ROOT.RooArgList(intercept, slope))

fitter.setTarget(tree, 'p', 'x')
fitter.setLikelihood(line, ROOT.RooArgList(intercept, slope))

fitter.fit(1)
